gold/nanomips.cc

```diff
@@ -193,6 +193,32 @@ is_forced_insn_length(typename elfcpp::Elf_types<size>::Elf_Addr offset,
   return false;
 }
 
+// Sym number associated with R_NANOMIPS_NONE relocation.
+// 0 If none.
+template<int size, bool big_endian>
+static inline unsigned int
+has_none_reloc(typename elfcpp::Elf_types<size>::Elf_Addr offset,
+                        size_t reloc_count,
+                        size_t relnum,
+                        const unsigned char* preloc)
+{
+  typedef typename elfcpp::Rela<size, big_endian> Reltype;
+  const int reloc_size = elfcpp::Elf_sizes<size>::rela_size;
+
+  preloc += reloc_size;
+  for (size_t i = relnum + 1; i < reloc_count; ++i, preloc += reloc_size)
```

Comment:
Note for future improvement: reloc list is probably pre-sorted and amenable to a faster search.